solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceBaseInputOperator.java

```diff
@@ -0,0 +1,319 @@
+  protected LinkedList<Long> inFlightMessageId = new LinkedList<Long>(); //keeps track of all in flight IDs since they are not necessarily sequential
+  protected transient ArrayBlockingQueue<BytesXMLMessage> arrivedTopicMessagesToProcess;
+
+  //protected transient com.solace.dt.operator.DTSolaceOperatorInputOutput.ArrayBlockingQueue<BytesXMLMessage> arrivedMessagesToProcess;
+
+  private transient ReconnectCallbackHandler rcHandler = new ReconnectCallbackHandler();
+
+  private transient CallbackMessageHandler cbHandler = new CallbackMessageHandler();
+
+  int spinMillis;
+
+  protected transient int reconnectRetryMillis = 0;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+
+    operatorId = context.getId();
+    logger.info("OperatorID from Base class: {}", operatorId);
+    spinMillis = context.getValue(com.datatorrent.api.Context.OperatorContext.SPIN_MILLIS);
+    factory = JCSMPFactory.onlyInstance();
+
+    //Required for HA and DR to try forever if set to "-1"
+    JCSMPChannelProperties channelProperties = (JCSMPChannelProperties)this.properties.getProperty(JCSMPProperties.CLIENT_CHANNEL_PROPERTIES);
+    channelProperties.setConnectRetries(Integer.parseInt(this.connectRetries));
+    channelProperties.setReconnectRetries(Integer.parseInt(this.reconnectRetries));
+
+    reconnectRetryMillis = channelProperties.getReconnectRetryWaitInMillis();
+
+
+    try {
+      session = factory.createSession(this.properties, null, new PrintingSessionEventHandler());
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+
+    //logger.debug("Properties Raw: \n{}", properties.toProperties());
+    logger.debug("Properties:\n" + properties.toString());
+    //logger.debug("\n===============================================\n");
+
+    idempotentStorageManager.setup(context);
+    lastCompletedWId = idempotentStorageManager.getLargestRecoveryWindow();
+    //logger.debug("++++++++++++++++++++Largest Completed: " + lastCompletedWId);
+
+
+  }
+
+  @Override
+  public void checkpointed(long arg0)
+  {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public void committed(long window)
+  {
+    try {
+      idempotentStorageManager.deleteUpTo(operatorId, window);
+    } catch (IOException e) {
+      DTThrowable.rethrow(e);
+    }
+
+  }
+
+  protected T processMessage(BytesXMLMessage message)
+  {
+    T tuple = convert(message);
+    if (tuple != null) {
+      emitTuple(tuple);
+    }
+    return tuple;
+  }
+
+
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+    try {
+      session.connect();
+      reliableConsumer = session.getMessageConsumer(rcHandler, cbHandler);
+      //consumer = getConsumer();
+      reliableConsumer.start();
+
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      consumer.stop();
+      clearConsumer();
```

Comment:
clearConsumer() is a no-op everywhere. Can we remove the definition and call unless there is a reason for it in the future?